wiki/PreventiveMaintenance.md

```diff
@@ -0,0 +1,64 @@
+ * parallel LineSegments. Moreover this function guarantees to return an intersection point if the LineSegments intersects. In case:
+ * - There is no intersection then std:nullopt is returned.
+ * - There is a single intersection point then that intersection point is returned.
+ * - There are multiple intersection points (infinitely many) which happens when the LineSegments overlap then it prefers to returns 1: start, 2: end, 3: 
+ *  line.start of which the first most option is selected that lies on both LineSegments. Note that if none of these points lie on both LineSegments and if both 
+ *  the LineSegments are parallel then the LineSegments cannot intersect.
+ *
+ * @param line The other (finite) LineSegment.
+ * @return std::nullopt if the LineSegments do not intersect. Otherwise return an intersection point.
+ */
+[[nodiscard]] std::optional<Vector2> intersects(const LineSegment &line) const;
+```
+Note that this documentation gives a clear description of what the code does, by explaining what happens for each situation. But beside that it also explain why this code works, by telling that the function knows how to deal with parallel & point cases and that you don't have to check if line.end is on the line if start, end and line.start are not on both lines.
+
+## Bad Example
+Below is an example how you should not document your code:
+```
+/**
+ * Calculates info for the defenders
+ */
+void calculateInfoForDefenders() noexcept;
+```
+The documentation basically tells the same as the name of the function. Hence the goal of documentation, to inform others about usage, is missing here. What kind of info is computed is totally unclear? What are requirements for calling this function is also missing? When should this function be called?
+
+# Design
+
+# Unit Testing
+## Why?
+Writing unit test cases for your code is very important, because
+- Even the simplest code can contain bugs and even when fully reasoned that the code works, it can still contain some bugs.
+- It encourage you to inspect your code better for bugs, because during the process of writing test you might already find bugs (without executing the tests).
+- It improves the trust in the code, since it is properly tested and therefore it is less likely that your code suddenly fails during the RoboCup.
+- It will help you find the location of bugs.
+
+## How?
+When writing unit test cases, your goal is to create unit test cases that might/will fail. So for example you should test the path planning algorithm for the case that the ball cannot be reached (because it is surrounded by enemy robots) and for LineSegments you should test LineSegments that are actually points (have an equal start position as end position). Using random unit test cases or just writing a few test cases will in most cases not found the most common type of bugs. Hence you should be more systematic in writing test cases. For this there are 2 types of approaches: black box testing and white box testing.
```

Comment:
```suggestion
When writing unit test cases, your goal is to create unit test cases that might/will fail. So for example you should test the path planning algorithm for the case that the ball cannot be reached (because it is surrounded by enemy robots) and for LineSegments you should test LineSegments that are actually points (have an equal start position as end position). Using random unit test cases or just writing a few test cases will in most cases not find the most common type of bugs. Hence you should be more systematic in writing test cases. For this there are 2 types of approaches: black box testing and white box testing.
```